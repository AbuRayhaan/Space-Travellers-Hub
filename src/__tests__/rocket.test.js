import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../components/rocket/RocketList';

const RocketProvider = () => (
  <Provider store={store}>
    <Rockets />
  </Provider>
);

describe('Test rockets components', () => {
  it('check for proper rendering', () => {
    const rocketItems = renderer.create(<RocketProvider />).toJSON();
    expect(rocketItems).toMatchSnapshot();
  });

  it('has a text in the DOM', () => {
    const { rocketList } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(rocketList.getElementsByClassName('rocketList').length).toBe(1);
  });
});
