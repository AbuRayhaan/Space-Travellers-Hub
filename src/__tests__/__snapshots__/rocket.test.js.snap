// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets components test check for proper rendering 1`] = `
<div
  className="rocketList"
/>
`;